prov/util/src/util_coll.c

```diff
@@ -1138,6 +1141,27 @@ static int util_coll_av_init(struct util_av *av)
 	return ret;
 }
 
+static int util_av_set_close(struct fid *fid)
+{
+	struct util_av_set *av_set;
+
+	av_set = container_of(fid, struct util_av_set, av_set_fid.fid);
+
+	ofi_atomic_dec32(&av_set->ref);
+	assert(ofi_atomic_get32(&av_set->ref) == 0);
```

Comment:
The assert can be converted into an if statement and return -FI_EBUSY if ref > 0.  Please add a comment above the dec32 along the lines of 'release reference taken by internal coll_mc'.

Reply:
I see. changed.